worlds/jakanddaxter/__init__.py

```diff
@@ -79,48 +90,63 @@ class JakAndDaxterWorld(World):
-                        if k in range(jak1_id, jak1_id + Scouts.fly_offset)},
-        "Scout Flies": {location_table[k]: k for k in location_table
-                        if k in range(jak1_id + Scouts.fly_offset, jak1_id + Specials.special_offset)},
-        "Specials": {location_table[k]: k for k in location_table
-                     if k in range(jak1_id + Specials.special_offset, jak1_id + Caches.orb_cache_offset)},
-        "Orb Caches": {location_table[k]: k for k in location_table
-                       if k in range(jak1_id + Caches.orb_cache_offset, jak1_id + Orbs.orb_offset)},
-        "Precursor Orbs": {location_table[k]: k for k in location_table
-                           if k in range(jak1_id + Orbs.orb_offset, jak1_max)},
-        "Trades": {location_table[k]: k for k in location_table
-                   if k in {Cells.to_ap_id(t) for t in {11, 12, 31, 32, 33, 96, 97, 98, 99, 13, 14, 34, 35, 100, 101}}},
+        "Power Cells": {cell_location_table[k] for k in cell_location_table},
+        "Scout Flies": {scout_location_table[k] for k in scout_location_table},
+        "Specials": {special_location_table[k] for k in special_location_table},
+        "Orb Caches": {cache_location_table[k] for k in cache_location_table},
+        "Precursor Orbs": {orb_location_table[k] for k in orb_location_table},
+        "Trades": {location_table[Cells.to_ap_id(k)] for k in
+                   {11, 12, 31, 32, 33, 96, 97, 98, 99, 13, 14, 34, 35, 100, 101}},
     }
 
+    # Functions and Variables that are Options-driven, keep them as instance variables here so that we don't clog up
+    # the seed generation routines with options checking. So we set these once, and then just use them as needed.
+    can_trade: Callable[[CollectionState, int, Optional[int]], bool]
+    orb_bundle_size: int = 0
+    orb_bundle_item_name: str = ""
+
+    def generate_early(self) -> None:
+        # For the fairness of other players in a multiworld game, enforce some friendly limitations on our options,
+        # so we don't cause chaos during seed generation. These friendly limits should **guarantee** a successful gen.
+        if self.multiworld.players > 1:
+            from .Rules import enforce_multiplayer_limits
+            enforce_multiplayer_limits(self.options)
+
+        # Verify that we didn't overload the trade amounts with more orbs than exist in the world.
+        # This is easy to do by accident even in a single-player world.
+        from .Rules import verify_orb_trade_amounts
+        verify_orb_trade_amounts(self.options)
+
+        # Cache the orb bundle size and item name for quicker reference.
+        if self.options.enable_orbsanity == EnableOrbsanity.option_per_level:
+            self.orb_bundle_size = self.options.level_orbsanity_bundle_size.value
+            self.orb_bundle_item_name = orb_item_table[self.orb_bundle_size]
+
+        if self.options.enable_orbsanity == EnableOrbsanity.option_global:
```

Comment:
They're mutually exclusive so you can use `elif`
```suggestion
        elif self.options.enable_orbsanity == EnableOrbsanity.option_global:
```